x/clock/abci.go

```diff
@@ -31,12 +33,19 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 		childCtx := ctx.WithGasMeter(sdk.NewGasMeter(p.ContractGasLimit))
 		_, err = k.GetContractKeeper().Sudo(childCtx, contract, message)
 		if err != nil {
+			errorExists = true
 			errorExecs[idx] = addr
-			continue
+
+			// Jail contract on error
+			err = k.JailContract(ctx, addr)
+			if err != nil {
+				log.Printf("[x/clock] Failed to Error Contract %s: %v", addr, err)
```

Comment:
I did this, but we should use a keeper logger instead where `logger log.Logger,` is passed into the keeper, we modify it as `logger = logger.With(log.ModuleKey, "x/"+poa.ModuleName)` and then save to the Keeper struct

```
func (k Keeper) Logger() log.Logger {
	return k.logger
}
```

then we can `k.GetLogger().Debug(...)` here

same below with `if errorExists { ...`

Reply:
Implemented logger in last commit